validator/js/chromeextension/background.js

```diff
@@ -95,6 +98,20 @@ function getNumberOfWarnings(errors) {
   return getErrorSeverityCounts(errors)['WARNING'];
 }
 
+/**
+ * Inspects ValidationResults to discover if in Dev Mode.
+ *
+ * @param {!Object<!ValidationResult>} validationResult
+ * @return {boolean}
+ */
+function isInDevMode(validationResult) {
+  if ((validationResult.errors.length == 1) &&
```

Comment:
I'm not sure this is what you actually want.

data-ampdevmode is enabled on the HTML tag, but doesn't guarantee that there are no additional errors. In fact, enabling it and doing nothing else doesn't suppress any errors, it just adds an error. It might not even be the first error (thought that's likely).

Instead, once the mode is enabled via the `html` tag, if the same attribute is then added to another other tag in the DOM, say `<html data-ampdevmode>...<body data-ampdevmode>...`, then the other tags with this attribute will have their errors, and the errors of their children, suppressed.

This isn't super obvious, but the reason for the feature was that you could hide errors for sections of the document, such as when a wordpress thing injects some js onto the document and wants to detect errors other than the ones from the injected js.

In other words, it's entirely possible that usage of ampdevmode might also come with more than 1 error. Maybe if there are non-devmode errors, you want to show this as non-devmode, I don't know what you actually want. If so, the naming of the fn is misleading and should get some comments to help the future reader.

Reply:
I was following the request from #24176 which states:

```
In such case, however, when there is only one validation error and it is due to being in debug mode, it would be very useful if the Validator extension would not show an error icon but rather a debug icon, similar to how it shows a warning icon when there are deprecation issues.
```

In my reading of that, the debug icon only shows when `data-ampdevmode` is on `html` and no other errors or any other errors are removed by having `data-ampdevmode` also on those tags.

Reply:
Elaborated in comment and included reference to issue 24176.

Reply:
Sounds good. I think my interpretation of this was based on the name of the function. The devmode behavior may be enabled in docs where isDevMode returns false, but for the definition you want for the extension, it's doing the right thing.

I'd lightly suggest tweaking the f'n name, but I don't have a strong opinion. Suggestions would be `displayAsDevMode` or `isDevModeOnlyError` or something like that.

Reply:
I disagree that isDevMode isn't correct for data-ampdevmode's intentions. If data-ampdevmode is not being used on other tags which are causing errors then it's not really in dev mode. But to move this forward I will rename it to `onlyErrorIsDevMode`.